DOC: Confound enhancement documentation (#1625)
poldracklab/fmriprep: docs/outputs.rst

and the motion-related components identified by
:abbr:`ICA (independent components analysis)`-:abbr:`AROMA (Automatic Removal Of Motion Artifacts)`
(if enabled) are indicated with ``aroma_motion_XX``.

Four separate CompCor decompositions are performed to compute noise components: one temporal
decomposition and three anatomical decompositions across three different noise ROIs: an eroded
white matter compartment, an eroded CSF compartment, and a combined mask derived from the union
of these. In general, only a subset of these decompositions should be used for further
denoising. The original Behzadi aCompCor implementation ([Behzadi2007]_) can be applied using
components from the combined ROI, while the more recent Muschelli implementation
([Muschelli2014]_) can be applied using the WM and CSF ROIs. To determine the provenance of
each component, consult the metadata file (see below).

All these confounds can be used to perform *scrubbing* and *censoring* of outliers,
in the subsequent first-level analysis when building the design matrix,
and in group level analysis. *Spike regressors* for outlier censoring can also be

> generated from within fMRIPrep using the command line options `--fd-spike-threshold`

Reviewer: ```suggestion
generated from within fMRIPrep using the command line options ``--fd-spike-threshold``
```

> and `--dvars-spike-threshold`. Spike regressors are stored in separate `motion_outlier_XX`

Reviewer: ```suggestion
and ``--dvars-spike-threshold``. Spike regressors are stored in separate ``motion_outlier_XX``
```

columns.

Each confounds data file will also have a corresponding metadata file (``~desc-confounds_regressors.json``).
Metadata files contain additional information about columns in the confounds TSV file: ::

  {
    "a_comp_cor_00": {
      "CumulativeVarianceExplained": 0.1081970825,
      "Mask": "combined",
      "Method": "aCompCor",
      "Retained": true,
      "SingularValue": 25.8270209974,
      "VarianceExplained": 0.1081970825
    },
    "dropped_0": {
      "CumulativeVarianceExplained": 0.5965809597,
      "Mask": "combined",
      "Method": "aCompCor",
      "Retained": false,
      "SingularValue": 20.7955177198,
      "VarianceExplained": 0.0701465624
    }
  }

For CompCor decompositions, entries include:

> - ``Method``: anatomical or temporal CompCor

Reviewer: ```suggestion

  - ``Method``: anatomical or temporal CompCor.
```

> - ``Mask``: denotes the ROI where the decomposition that generated the component was performed: ``CSF``, ``WM``, or ``combined`` for anatomical CompCor

Reviewer: ```suggestion
  - ``Mask``: denotes the ROI where the decomposition that generated the component
    was performed: ``CSF``, ``WM``, or ``combined`` for anatomical CompCor.

> - ``SingularValue``: Singular value of the component

Reviewer: ```suggestion
  - ``SingularValue``: Singular value of the component.
```

> - ``VarianceExplained``: fraction of variance explained by the component across the decomposition ROI mask

Reviewer: ```suggestion
  - ``VarianceExplained``: the fraction of variance explained by the component across the decomposition ROI mask.
```

> - ``CumulativeVarianceExplained``: total fraction of variance explained by the component and all preceding components

Reviewer: ```suggestion
  - ``CumulativeVarianceExplained``: the total fraction of variance explained by this particular component
    and all preceding components.
```

> - ``Retained``: Indicates whether the component was saved in ``desc-confounds_regressors.tsv`` for use in denoising. Entries that are not saved in the data file for denoising are still stored in metadata with the ``dropped`` prefix.

Reviewer: ```suggestion
  - ``Retained``: Indicates whether the component was saved in ``desc-confounds_regressors.tsv``
    for use in denoising.
    Entries that are not saved in the data file for denoising are still stored in metadata with the
    ``dropped`` prefix.

```

    in cortical gray matter regions (blue), subcortical gray matter (orange),
    cerebellum (green) and the union of white-matter and CSF compartments (red).

Noise components computed during each CompCor decomposition are evaluated according
to the fraction of variance that they explain across the nuisance ROI.

> This is used by fMRIPrep to determine whether each component should be saved for

Reviewer: ```suggestion
This is used by *fMRIPrep* to determine whether each component should be saved for
```

use in denoising operations: a component is saved if it contributes to explaining

> the top 50 percent of variance in the nuisance ROI. fMRIPrep can be configured to

Reviewer: ```suggestion
the top 50 percent of variance in the nuisance ROI.
*fMRIPrep* can be configured to
```

save all components instead using the command line option ``--return-all-components``.

> FMRIPrep reports include a plot of the cumulative variance explained by each

Reviewer: ```suggestion
*fMRIPrep* reports include a plot of the cumulative variance explained by each
```

component, ordered by descending singular value.

.. figure:: _static/sub-01_task-rest_compcor.svg
    :scale: 100%

    The figure displays the cumulative variance explained by components for each
    of four CompCor decompositions (left to right: anatomical CSF mask, anatomical

>     white matter mask, anatomical combined mask, temporal). The number of

Reviewer: ```suggestion
    white matter mask, anatomical combined mask, temporal).
    The number of components is plotted on the abscissa and
```

>     components is plotted on the abscissa and the cumulative variance explained

Reviewer: ```suggestion
    the cumulative variance explained on the ordinate.
```

>     on the ordinate. Dotted lines indicate the minimum number of components

Reviewer: ```suggestion
    Dotted lines indicate the minimum number of components necessary
```

>     necessary to explain 50%, 70%, and 90% of variance in the nuisance mask. By

Reviewer: ```suggestion
    to explain 50%, 70%, and 90% of the variance in the nuisance mask.
```

>     default, only the components that explain the top 50% of variance are saved.

Reviewer: ```suggestion
    By default, only the components that explain the top 50% of the variance
    are saved.
```

Reviewer: Is it possible to change this threshold of 50% with some flag or some other mechanism?

Author: Not possible right now, but it wouldn't take much effort to change the `--return-all-components` option to `--compcor-variance-threshold` with this behaviour.

> Also included is a plot of correlations among confound regressors. This can be

Reviewer: Please start new sentences with newline (sorry I'll stop providing suggestions for this)
